tests/end2end/test_invocations.py

```diff
@@ -177,3 +183,83 @@ def test_optimize(request, quteproc_new, capfd, level):
     # Waiting for quit to make sure no other warning is emitted
     quteproc_new.send_cmd(':quit')
     quteproc_new.wait_for_quit()
+
+
+def test_version(request):
+    """Test invocation with --version argument."""
+    args = ['--version'] + _base_args(request.config)
+    # can't use quteproc_new here because it's confused by
+    # early process termination
+    proc = QuteProc(request)
+    proc.proc.setProcessChannelMode(QProcess.SeparateChannels)
+
+    try:
+        proc.start(args)
+        proc.wait_for_quit()
+    except ProcessExited:
+        assert proc.proc.exitStatus() == QProcess.NormalExit
+        pass
+    else:
+        assert False
+
+    proc.proc.setReadChannel(QProcess.StandardOutput)
+    lines = []
+    while proc.proc.canReadLine():
+        line = proc.proc.readLine()
+        lines.append(bytes(line).decode('utf-8').rstrip('\r\n'))
```

Comment:
You're right, in fact, it can all be condensed to a simple `output = bytes(proc.proc.readAllStandardOutput()).decode('utf-8')`